cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1108,47 +1126,18 @@ private ApplicationDetail enforceApplicationDetailAccess(ApplicationId appId,
   }
 
   // get filter for app specs by artifact name and version. if they are null, it means don't filter.
-  private Predicate<ApplicationDetail> getAppPredicate(Set<String> artifactNames, @Nullable String artifactVersion) {
+  private List<ApplicationFilter> getAppPredicates(Set<String> artifactNames, @Nullable String artifactVersion) {
     if (artifactNames.isEmpty() && artifactVersion == null) {
-      return r -> true;
+      return Collections.emptyList();
     } else if (artifactNames.isEmpty()) {
```

Comment:
no need for `else` already returned above, same for the following lines.

Reply:
Done. Changed to list builder